File: Monotonic-Array.py
class Solution(object):
    def isMonotonic(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        s, d = True, True
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                d = False
            if nums[i] < nums[i-1]:
                s = False
        return s or d

File: Monotonic-Array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        boolean s = true; 
        boolean d = true; 
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) d = false;
            if (nums[i] < nums[i - 1]) s = false;
        }
        return s || d;
    }
}
